### PR TITLE
Add AGENT_TOLERATION_KEY

### LINKED ISSUE #1392: Add AGENT_TOLERATION_KEY to rook-operator to allow tolerations for specific keys only
Is this a bug report or feature request?

* Feature Request

**Feature Request**

Are there any similar features already existing:
Yes AGENT_TOLERATION, which would tolerate taints with *all* keys 

What should the feature do:
Add an optional ENV var to allow setting a specific key for the toleration

What would be solved through this feature:
i.e. rook-agent running on master nodes without need

Does this have an impact on existing features:
don't think so.

### PR DESCRIPTION
**Pull request by christianhuening**
Description of your changes:

Which issue is resolved by this Pull Request:
Resolves #1392

Checklist:
- [x] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.

### CODE REVIEW

File: pkg/operator/agent/agent.go [hunk 1]
```diff
@@ -36,10 +36,11 @@ import (
 )
 
 const (
-	agentDaemonsetName          = "rook-agent"
-	flexvolumePathDirEnv        = "FLEXVOLUME_DIR_PATH"
-	flexvolumeDefaultDirPath    = "/usr/libexec/kubernetes/kubelet-plugins/volume/exec/"
-	agentDaemonsetTolerationEnv = "AGENT_TOLERATION"
+	agentDaemonsetName          		= "rook-agent"
```
Comment (bassam) [inline pkg/operator/agent/agent.go]: This failed go.fmt

File: Gopkg.lock [hunk 2]
```diff
@@ -270,12 +270,6 @@
   packages = [".","xfs"]
   revision = "a6e9df898b1336106c743392c48ee0b71f5c4efa"
 
-[[projects]]
-  branch = "master"
-  name = "github.com/rook/operator-kit"
-  packages = ["."]
```
Comment (bassam) [inline Gopkg.lock]: I dont follow why this was removed

**Comment by bassam** [comment]
Can you please squash the commits?

**Comment by christianhuening** [comment]
i fuddled up the PR, will create a new one

**Pull request closed without merging**